"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/jumpserver/api.py:433:29:433:29""]] flows to this location and may be exposed to an external user.
[[""Stack trace information""|""relative:///apps/jumpserver/api.py:450:29:450:29""]] flows to this location and may be exposed to an external user.","/apps/jumpserver/api.py","465","25","465","28"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/terminal/api/applet/applet.py:50:32:50:32""]] flows to this location and may be exposed to an external user.","/apps/terminal/api/applet/applet.py","51","35","51","85"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/terminal/models/applet/applet.py:111:29:111:29""]] flows to this location and may be exposed to an external user.","/apps/terminal/models/applet/applet.py","112","35","112","88"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/authentication/backends/oidc/backends.py:164:29:164:29""]] flows to this location and may be exposed to an external user.","/apps/authentication/backends/oidc/backends.py","168","30","168","34"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/authentication/backends/oidc/backends.py:205:33:205:33""]] flows to this location and may be exposed to an external user.","/apps/authentication/backends/oidc/backends.py","209","34","209","38"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/common/drf/parsers/base.py:193:29:193:29""]] flows to this location and may be exposed to an external user.","/apps/common/drf/parsers/base.py","195","30","195","64"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/jumpserver/views/celery_flower.py:21:25:21:25""]] flows to this location and may be exposed to an external user.","/apps/jumpserver/views/celery_flower.py","24","33","24","35"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/authentication/api/connection_token.py:241:30:241:30""]] flows to this location and may be exposed to an external user.","/apps/authentication/api/connection_token.py","242","34","242","50"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/settings/api/email.py:69:29:69:29""]] flows to this location and may be exposed to an external user.","/apps/settings/api/email.py","71","29","71","45"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/common/serializers/fields.py:260:30:260:30""]] flows to this location and may be exposed to an external user.","/apps/common/serializers/fields.py","261","47","261","47"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:612:42:612:42""]] flows to this location and may be exposed to an external user.
[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:614:33:614:33""]] flows to this location and may be exposed to an external user.
[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:616:53:616:53""]] flows to this location and may be exposed to an external user.
[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:618:57:618:57""]] flows to this location and may be exposed to an external user.
[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:620:48:620:48""]] flows to this location and may be exposed to an external user.
[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:622:29:622:29""]] flows to this location and may be exposed to an external user.","/apps/settings/api/ldap.py","49","25","49","27"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:658:42:658:42""]] flows to this location and may be exposed to an external user.
[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:660:50:660:50""]] flows to this location and may be exposed to an external user.
[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:662:49:662:49""]] flows to this location and may be exposed to an external user.
[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:664:29:664:29""]] flows to this location and may be exposed to an external user.","/apps/settings/api/ldap.py","93","25","93","27"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/settings/api/ldap.py:158:29:158:29""]] flows to this location and may be exposed to an external user.","/apps/settings/api/ldap.py","160","34","160","37"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:612:42:612:42""]] flows to this location and may be exposed to an external user.
[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:614:33:614:33""]] flows to this location and may be exposed to an external user.
[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:616:53:616:53""]] flows to this location and may be exposed to an external user.
[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:618:57:618:57""]] flows to this location and may be exposed to an external user.
[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:620:48:620:48""]] flows to this location and may be exposed to an external user.
[[""Stack trace information""|""relative:///apps/settings/utils/ldap.py:622:29:622:29""]] flows to this location and may be exposed to an external user.","/apps/settings/api/ldap.py","171","38","171","49"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/settings/api/ldap.py:219:29:219:29""]] flows to this location and may be exposed to an external user.","/apps/settings/api/ldap.py","220","29","220","45"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/settings/api/ldap.py:246:29:246:29""]] flows to this location and may be exposed to an external user.","/apps/settings/api/ldap.py","248","34","248","48"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/common/serializers/mixin.py:152:42:152:42""]] flows to this location and may be exposed to an external user.","/apps/common/serializers/mixin.py","159","35","159","40"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/notifications/api/notifications.py:115:29:115:29""]] flows to this location and may be exposed to an external user.","/apps/notifications/api/notifications.py","132","25","132","45"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/settings/api/sms.py:161:29:161:29""]] flows to this location and may be exposed to an external user.","/apps/settings/api/sms.py","164","50","164","53"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/authentication/api/token.py:46:29:46:29""]] flows to this location and may be exposed to an external user.","/apps/authentication/api/token.py","47","29","47","45"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/settings/api/vault.py:36:29:36:29""]] flows to this location and may be exposed to an external user.","/apps/settings/api/vault.py","44","46","44","57"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/authentication/backends/saml2/views.py:204:40:204:44""]] flows to this location and may be exposed to an external user.","/apps/authentication/backends/saml2/views.py","206","33","206","37"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///apps/authentication/backends/saml2/views.py:249:40:249:44""]] flows to this location and may be exposed to an external user.","/apps/authentication/backends/saml2/views.py","251","33","251","37"
"Accepting unknown SSH host keys when using Paramiko","Accepting unknown host keys can allow man-in-the-middle attacks.","error","Setting missing host key policy to AutoAddPolicy may be unsafe.","/apps/ops/ansible/modules_utils/custom_common.py","35","9","35","73"
"Accepting unknown SSH host keys when using Paramiko","Accepting unknown host keys can allow man-in-the-middle attacks.","error","Setting missing host key policy to AutoAddPolicy may be unsafe.","/apps/assets/automations/ping_gateway/manager.py","28","9","28","68"
"Accepting unknown SSH host keys when using Paramiko","Accepting unknown host keys can allow man-in-the-middle attacks.","error","Setting missing host key policy to AutoAddPolicy may be unsafe.","/apps/assets/automations/ping_gateway/manager.py","30","9","30","67"
"Clear-text logging of sensitive information","Logging sensitive information without encryption or hashing can expose it to an attacker.","error","This expression logs [[""sensitive data (secret)""|""relative:///apps/accounts/automations/change_secret/manager.py:100:47:100:62""]] as clear text.","/apps/accounts/automations/change_secret/manager.py","99","19","101","13"
"Clear-text storage of sensitive information","Sensitive information stored without encryption or hashing can expose it to an attacker.","error","This expression stores [[""sensitive data (secret)""|""relative:///apps/accounts/models/base.py:155:25:155:35""]] as clear text.","/apps/accounts/models/base.py","155","25","155","35"
"Clear-text storage of sensitive information","Sensitive information stored without encryption or hashing can expose it to an attacker.","error","This expression stores [[""sensitive data (secret)""|""relative:///apps/assets/automations/base/manager.py:108:20:108:67""]] as clear text.","/apps/assets/automations/base/manager.py","101","21","101","27"
"Clear-text storage of sensitive information","Sensitive information stored without encryption or hashing can expose it to an attacker.","error","This expression stores [[""sensitive data (secret)""|""relative:///apps/assets/automations/base/manager.py:148:35:148:40""]] as clear text.","/apps/assets/automations/base/manager.py","157","25","157","30"
"Use of weak cryptographic key","Use of a cryptographic key that is too small may allow the encryption to be broken.","error","Creation of an RSA key uses [[""1024""|""relative:///apps/common/utils/crypto.py:251:25:251:28""]] bits, which is below 2048 and considered breakable.","/apps/common/utils/crypto.py","256","11","256","48"
"Use of a broken or weak cryptographic hashing algorithm on sensitive data","Using broken or weak cryptographic hashing algorithms can compromise security.","warning","[[""Sensitive data (secret)""|""relative:///apps/common/sdk/sms/cmpp2.py:42:31:42:39""]] is used in a hashing algorithm (MD5) that is insecure.
[[""Sensitive data (secret)""|""relative:///apps/common/sdk/sms/cmpp2.py:49:21:49:45""]] is used in a hashing algorithm (MD5) that is insecure.
[[""Sensitive data (secret)""|""relative:///apps/common/sdk/sms/cmpp2.py:250:66:250:79""]] is used in a hashing algorithm (MD5) that is insecure.","/apps/common/sdk/sms/cmpp2.py","53","39","53","58"
"Use of a broken or weak cryptographic hashing algorithm on sensitive data","Using broken or weak cryptographic hashing algorithms can compromise security.","warning","[[""Sensitive data (secret)""|""relative:///apps/common/utils/encode.py:225:20:225:36""]] is used in a hashing algorithm (MD5) that is insecure.","/apps/common/utils/encode.py","220","28","220","47"
"Use of a broken or weak cryptographic hashing algorithm on sensitive data","Using broken or weak cryptographic hashing algorithms can compromise security.","warning","[[""Sensitive data (secret)""|""relative:///apps/assets/automations/base/manager.py:148:35:148:40""]] is used in a hashing algorithm (MD5) that is insecure.","/apps/assets/automations/base/manager.py","149","30","149","51"
"Use of a broken or weak cryptographic hashing algorithm on sensitive data","Using broken or weak cryptographic hashing algorithms can compromise security.","warning","[[""Sensitive data (secret)""|""relative:///apps/common/sdk/im/utils.py:11:21:11:31""]] is used in a hashing algorithm (MD5) that is insecure.","/apps/common/sdk/im/utils.py","14","16","14","35"
"URL redirection from remote source","URL redirection based on unvalidated user input may cause redirection to malicious web sites.","error","Untrusted URL redirection depends on a [[""user-provided value""|""relative:///apps/authentication/views/login.py:220:19:220:25""]].","/apps/authentication/views/login.py","228","29","228","40"
"URL redirection from remote source","URL redirection based on unvalidated user input may cause redirection to malicious web sites.","error","Untrusted URL redirection depends on a [[""user-provided value""|""relative:///apps/authentication/views/login.py:301:16:301:27""]].","/apps/authentication/views/login.py","320","20","320","59"
"URL redirection from remote source","URL redirection based on unvalidated user input may cause redirection to malicious web sites.","error","Untrusted URL redirection depends on a [[""user-provided value""|""relative:///apps/authentication/views/login.py:301:16:301:27""]].","/apps/authentication/views/login.py","322","20","322","63"
"URL redirection from remote source","URL redirection based on unvalidated user input may cause redirection to malicious web sites.","error","Untrusted URL redirection depends on a [[""user-provided value""|""relative:///apps/authentication/views/login.py:301:16:301:27""]].","/apps/authentication/views/login.py","324","20","324","67"
"URL redirection from remote source","URL redirection based on unvalidated user input may cause redirection to malicious web sites.","error","Untrusted URL redirection depends on a [[""user-provided value""|""relative:///apps/authentication/mixins.py:539:16:539:27""]].","/apps/authentication/mixins.py","542","25","542","33"
"URL redirection from remote source","URL redirection based on unvalidated user input may cause redirection to malicious web sites.","error","Untrusted URL redirection depends on a [[""user-provided value""|""relative:///apps/jumpserver/views/other.py:32:19:32:25""]].","/apps/jumpserver/views/other.py","34","41","34","51"
"URL redirection from remote source","URL redirection based on unvalidated user input may cause redirection to malicious web sites.","error","Untrusted URL redirection depends on a [[""user-provided value""|""relative:///apps/users/views/profile/password.py:45:19:45:30""]].
Untrusted URL redirection depends on a [[""user-provided value""|""relative:///apps/users/views/profile/password.py:46:20:46:31""]].","/apps/users/views/profile/password.py","42","25","42","46"
